frontend/device/generic/device.lua

```diff
@@ -116,8 +117,14 @@ end
 -- ONLY used for Kobo and PocketBook devices
 function Device:onPowerEvent(ev)
     local Screensaver = require("ui/screensaver")
+    local network_manager = require("ui/network/manager")
     if (ev == "Power" or ev == "Suspend") and not self.screen_saver_mode then
         self.powerd:beforeSuspend()
+        self.wifi_was_on = network_manager:getWifiStatus()
```

Comment:
Let's first agree on what the user experience should be. IMHO, KOReader should remember a user's network preference and try its best to maintain the state of the network on resume.

Doing the check before suspend each time makes no guarantee on remembering the user's preference. What it actually does is remembering the network connectivity state that the reader sees before suspend. Consider the following situation:

1. user connected to a WIFI (at home) from the network manager
1. user went outside of the WIFI range while reading, then suspended the reader
1. user went back home and resumed the reader

Now, should KOReader resume the WIFI or not? I think it makes more sense to reconnect to the home WIFI. However, if you only do the wifi state check before suspend, `network_manager:getWifiStatus()` will return false while the user is outside of the home WIFI range. As a result the network will not be resumed after the user got back home.

From a performance perspective, `network_manager:getWifiStatus()` is a very expensive blocking function call. While I was developing the network manager, it constantly freeze the whole UI for 10+ seconds after a manual WIFI disconnect. I fixed this by forcing the device to release IP so the DNS lookup can fail immediately. But this is just one edge-case. There are many other cases when a network issue will cause this lookup to block. So if we use it in the suspend handler, it might block the suspend action for a long time too.

The ideal case is to remember this user preference whenever a user manually connects to a WIFI through KOReader. Write the state to disk. Then read this state into network manager every time KOReader starts.